src/libs/elektra/keyset.c

```diff
@@ -2600,6 +2648,7 @@ int ksInit (KeySet * ks)
 	ks->size = 0;
 	ks->alloc = 0;
 	ks->flags = 0;
+	ks->refs = 1; // there is always at least one active reference
```

Comment:
This should be analogous to Key (which starts with 0), otherwise it gets very confusing for API users.

Reply:
Key starts with 0, because it originally only counted references from KeySets. In that case 0 makes sense. A Key that was just created is not part of a KeySet.

However, this is a general reference counter. Here we count all references, and there is always at least one of those. Because unreferenced memory is safe to deallocate.

Another benefit of starting with 1 is that 0 can be used as an error value. So it's easier to use an unsigned type.

Apart from these reasons, I would argue starting a reference count with 0 is more confusing to API users (unless it only counts a specific type of references, e.g. from KeySet to Key).

Reply:
> Key starts with 0, because it originally only counted references from KeySets. In that case 0 makes sense. A Key that was just created is not part of a KeySet.

> However, this is a general reference counter.

Then what about making the Key refcounter also a general reference counter? Sounds less confusing to me.